base/poco/Net/include/Poco/Net/MessageHeader.h

```diff
@@ -157,12 +177,14 @@ namespace Net
         enum Limits
         /// Limits for basic sanity checks when reading a header
         {
-            MAX_NAME_LENGTH = 256,
-            MAX_VALUE_LENGTH = 8192,
+            DFL_NAME_LENGTH_LIMIT = 256,
+            DFL_VALUE_LENGTH_LIMIT = 8192,
```

Comment:
The default maximum length of a HTTP header's value was `8K` which was not enough for some S3 API headers. So this PR increases it to `512K` and makes it configurable.